tests/cases/compiler/renamingDestructuredPropertyInFunctionType.ts

```diff
@@ -0,0 +1,65 @@
+// @target: es2015
+// @declaration: true
+// GH#37454, GH#41044
+
+type O = { a?: string; b: number; c: number; };
+type F1 = (arg: number) => any; // OK
+type F2 = ({ a: string }: O) => any; // Error
+type F3 = ({ a: string, b, c }: O) => any; // Error
+type F4 = ({ a: string }: O) => any; // Error
+type F5 = ({ a: string, b, c }: O) => any; // Error
+type F6 = ({ a: string }) => typeof string; // OK
+type F7 = ({ a: string, b: number }) => typeof number; // Error
+type F8 = ({ a, b: number }) => typeof number; // OK
+type F9 = ([a, b, c]) => void; // Error
+type F10 = ({ "a": string }) => void; // Error
+type F11 = ({ 2: string }) => void; // Error
+type F12 = ({ ["a"]: string }: O) => void; // Error
+type F13 = ({ [2]: string }) => void; // Error
+
+type G1 = new (arg: number) => any; // OK
+type G2 = new ({ a: string }: O) => any; // Error
+type G3 = new ({ a: string, b, c }: O) => any; // Error
+type G4 = new ({ a: string }: O) => any; // Error
+type G5 = new ({ a: string, b, c }: O) => any; // Error
+type G6 = new ({ a: string }) => typeof string; // OK
+type G7 = new ({ a: string, b: number }) => typeof number; // Error
+type G8 = new ({ a, b: number }) => typeof number; // OK
+type G9 = new ([a, b, c]) => void; // Error
+type G10 = new ({ "a": string }) => void; // Error
+type G11 = new ({ 2: string }) => void; // Error
+type G12 = new ({ ["a"]: string }: O) => void; // Error
+type G13 = new ({ [2]: string }) => void; // Error
```

Comment:
Can you add a comment explaining that `string` has to be retained in declaration emit even though it’s an error, because eliding it would leave invalid syntax (since the property names aren’t identifiers)?